llvm/lib/Target/VE/veintrin.py

```diff
@@ -970,19 +970,34 @@ def Inst3divbys(self, opc, name, instName, subop, ty):
         O = self.addMask(O)
         self.Def(opc, instName, subop, name, O, "{0} = {1} / {2}")
 
-    def Logical(self, opc, name, instName, expr):
+    def Logical2u(self, opc, name, instName):
+        O_u64 = [[VX(T_u64), VZ(T_u64)]]
+        O_u32 = [[VX(T_u32), VZ(T_u32)]]
+        O_pu32 = [[VX(T_u32), VZ(T_u32)]]
+
+        O_u64 = self.addMask(O_u64)
+        O_u32 = self.addMask(O_u32)
+        O_pu32 = self.addMask(O_pu32, VM512)
+
+        self.Def(opc, instName, "", name, O_u64).noTest()
+        self.Def(opc, instName, "p", "p" + name + ".lo", O_u32).noTest()
+        self.Def(opc, instName, "p", "p" + name + ".up", O_u32).noTest()
+        self.Def(opc, instName, "p", "p" + name, O_pu32).noTest()
+        
+    def Logical3u(self, opc, name, instName, expr):
         O_u32_vsv = [VX(T_u32), SY(T_u64), VZ(T_u32)]
 
-        Args = [Args_vvv(T_u64), Args_vsv(T_u64)]
-        Args = self.addMask(Args)
+        O_u64 = [Args_vvv(T_u64), Args_vsv(T_u64)]
+        O_u64 = self.addMask(O_u64)
 
-        ArgsP = [Args_vvv(T_u32), O_u32_vsv]
-        ArgsP = self.addMask(ArgsP, VM512)
+        O_pu32 = [Args_vvv(T_u32), O_u32_vsv]
+        O_pu32 = self.addMask(O_pu32, VM512)
 
-        self.Def(opc, instName, "", name, Args, expr)
-        #self.Def(opc, instName, "p", "p"+name+".lo", ArgsP, expr).noTest()
-        #self.Def(opc, instName, "p", "p"+name+".up", ArgsP, expr).noTest()
-        self.Def(opc, instName, "p", "p"+name, ArgsP, expr)
+        self.Def(opc, instName, "", name, O_u64, expr)
+        # self.Def(opc, instName, "p", "p"+name+".lo", O_pu32, expr).noTest()
+        # self.Def(opc, instName, "p", "p"+name+".up", O_pu32, expr).noTest()
+        self.Def(opc, instName, "p", "p" + name, O_pu32, expr)
+        
```

Comment:
Please remove this line (modify two empty lines into single empty line and remove spaces from empty lines)